pkg/apiserver/webhooks.go

```diff
@@ -296,72 +293,35 @@ func watchCertificates(directory, certFile, keyFile string) {
 			}
 		}
 	}()
-	// start the watcher with the dir to watch
-	err = watcher.Add(directory)
-	if err != nil {
-		klog.Fatalf("Error in running watcher: %v", err)
-	}
-
 	<-done
 }
-// return current cert
-func getCertificate(*tls.ClientHelloInfo) (*tls.Certificate, error) {
-	mu.Lock()
-	defer mu.Unlock()
-	return &cert, nil
-}
 
 func runWebhookServer(certStorageDir string, useCertManWebhook bool) error {
 	certFile := filepath.Join(certStorageDir, "tls.crt")
 	keyFile := filepath.Join(certStorageDir, "tls.key")
-	// load the cert for the first time
-
+	cert, err := loadCertificate(certFile, keyFile)
+	if err != nil {
+		klog.Errorf("failed to load certificate: %v", err)
+		return err
+	}
 	if useCertManWebhook {
-		_, err := loadCertificate(certFile, keyFile)
-		if err != nil {
-			klog.Errorf("failed to load certificate: %v", err)
-		}
-		// Start watching the certificate files for changes
-		// watch for changes in directory where secret is mounted
 		go watchCertificates(certStorageDir, certFile, keyFile)
```

Comment:
Do we need to run this in a separate go routine? If so, I suggest setting up the watcher separately and catching if it fails to avoid running the anonymous func for the watch.